packages/wallet-sdk/src/CoinbaseWalletProvider.ts

```diff
@@ -43,77 +41,46 @@ export class CoinbaseWalletProvider extends EventEmitter implements ProviderInte
     this.signer = signerType ? this.initSigner(signerType) : null;
   }
 
-  public async request<T>(args: RequestArguments): Promise<T> {
+  public async request(args: RequestArguments): Promise<unknown> {
     try {
       checkErrorForInvalidRequestArgs(args);
-      return (await this.handleRequest(args)) as T;
+      switch (args.method) {
+        // set up signer and return accounts
+        case 'eth_requestAccounts': {
```

Comment:
set up signer and return accounts

Reply:
might be easier to read with whitespace hidden
https://github.com/coinbase/coinbase-wallet-sdk/pull/1344/files?diff=split&w=1#diff-bb3b1961b89bdeabb2d5abe40c8646da1dda20ca67ded4d932c32002e76d70da